charts/prometheus/values.yaml

```diff
@@ -29,78 +29,100 @@ tolerations: []
+      target_timestamp_timezone:
+      component_field:
+      # Default component name if component_field is not set or field value is empty.
+      default_component_name:
+      # if no instance given, will set to 'Application'. Can also set instance name from multiple fields which separated by commas. Ex: instance_name_part1,instance_name_part2.
+      instance_field:
+      # field names used to get host id
+      dynamic_host_field:
+      # instance_whitelist is a regex string used to define which instances will be filtered.
+      instance_whitelist:
+      # Field name of device. Can also set device name from multiple fields which separated by commas. Ex: device_name_part1,device_name_part2.
+      device_field:
+      # The connector for build `instance_field` or `device_field` from multiple fields. Default is `-`.
+      instance_connector:
+      # number of thread to used in the pool, default is 20
+      thread_pool:
+      # number of the worker processes, default is count_of_cpu * 4
+      processes:
+      # minutes the main process wait for workers to finish, default is 5
+      timeout:
+      ## proxy
+      agent_http_proxy:
+      agent_https_proxy:
 
-  ## metrics
-  # Prometheus query used to query all the needed metrics. If not set, it will use `{__name__=~".+"}` to query all metrics.
-  prometheus_query:
-  # metric's name field, Multiple fields are separated by commas. EX: __name__, job, the metric name =  {__name__}_{job}
-  metrics_name_field:
-  # history data time config. If these options is set, the agent will query results by time range
-  # time range, Example: 2020-04-14 00:00:00,2020-04-15 00:00:00
-  his_time_range:
+    insightfinder:
+      user_name:
+      license_key:
+      token:
+      # Name of the project created in the InsightFinder UI, If this project is not exist, agent will create it automatically.
+      project_name:
+      # Name of system owned by project. If project_name is not exist in InsightFinder, agent will create a new system automatically from this field or project_name.
+      system_name:
+      # metric, metricreplay, log, logreplay, alert, alertreplay, incident, incidentreplay, deployment, deploymentreplay, trace, tracereplay
+      project_type:
+      # Set to `YES` if project is container.
+      containerize:
+      sampling_interval:
+      run_interval:
+      # what size to limit chunks sent to IF to, as kb
+      chunk_size_kb:
+      if_url:
+      if_http_proxy:
+      if_https_proxy:
 
-  # raw, rawtail, csv, csvtail, xls, xlsx, json, jsontail, avro, or xml
-  # *tail formats keep track of the current file & position + completed files in [state] below
-  data_format: json
+    # A list of prometheus query which is converted to json array in the file of prometheus_query_json.
+    # Reference the comment of prometheus_query_json for the fields.
+    queries:
+      # - query:
+      #   metric_name:
+      #   instance_fields:
```

Comment:
@bdulive Can we make it so that if this section is given/has data, it automatically creates a prometheus_query.json

Reply:
ok. I can make it.